src/mmw/js/src/draw/views.js

```diff
@@ -340,24 +243,46 @@ var AoIUploadView = Marionette.ItemView.extend({
     template: aoiUploadTmpl,
 
     ui: {
-        'selectFileInput': '#draw-tool-file-upload-input',
-        'selectFileButton': '#draw-tool-file-upload-button',
+        drawToolButton: '.draw-tool-button',
+        selectFileInput: '#draw-tool-file-upload-input',
+        selectFileButton: '#draw-tool-file-upload-button',
+        resetDrawButton: '.reset-draw-button'
     },
 
     events: {
         'dragenter': 'stopEvents',
         'dragover': 'stopEvents',
         'click @ui.selectFileButton': 'onSelectFileButtonClick',
         'change @ui.selectFileInput': 'selectFile',
+        'click @ui.drawToolButton': 'selectDrawToolItem',
+        'click @ui.resetDrawButton': 'reset'
+    },
+
+    modelEvents: {
+        change: 'render'
+    },
+
+    initialize: function() {
+        this.id = aoiUpload;
     },
 
-    onAttach: function() {
+    reset: function() {
+        this.model.reset();
+    },
+
+    selectDrawToolItem: function() {
+        this.model.selectDrawToolItem(this.id, this.id);
+    },
+
+    onRender: function() {
         // Using jQuery to bind this event prevents the file
         // data from being attached to the event object,
         // so we behind it manually once the view has been rendered
         // and attached to the DOM.
         var dropbox = document.getElementById("dropbox");
-        dropbox.addEventListener("drop", _.bind(this.drop, this), false);
+        if (dropbox) {
+            dropbox.addEventListener("drop", _.bind(this.drop, this), false);
+        }
```

Comment:
This is there because the dropbox element isn't present on each render because it's inside a conditional in the template. So I don't think this should be considered an error condition.